rust/pact_verifier_cli/src/main.rs

```diff
@@ -268,9 +268,30 @@ fn pact_source(matches: &ArgMatches) -> Vec<PactSource> {
           let auth = matches.value_of("user").map(|user| {
             HttpAuth::User(user.to_string(), matches.value_of("password").map(|p| p.to_string()))
           });
-          PactSource::BrokerUrl(name, s!(v), auth, vec![])
+          let pending = matches.is_present("enable-pending");
+          let wip = match matches.value_of("include-wip-pacts-since") {
+            Some(wip) => Some(wip.to_string()),
+            None => None,
+          };
+          let consumer_version_tags = matches.values_of("consumer-version-tags")
+            .map_or_else(|| vec![], |tags| consumer_tags_to_selectors(tags.map(|tag| tag).collect()));
+          let provider_tags = matches.values_of("provider-tags")
+            .map_or_else(|| vec![], |tags| tags.map(|tag| tag.to_string()).collect());
+
+          PactSource::BrokerWithDynamicConfiguration(name, s!(v), pending, wip, provider_tags, consumer_version_tags, auth, vec![])
         } else if matches.is_present("token") {
-          PactSource::BrokerUrl(s!(matches.value_of("provider-name").unwrap()), s!(v),
+          let name = matches.value_of("provider-name").unwrap().to_string();
+          let pending = matches.is_present("enable-pending");
+          let wip = match matches.value_of("include-wip-pacts-since") {
+            Some(wip) => Some(wip.to_string()),
+            None => None,
+          };
+          let consumer_version_tags = matches.values_of("consumer-version-tags")
+          .map_or_else(|| vec![], |tags| consumer_tags_to_selectors(tags.map(|tag| tag).collect()));
```

Comment:
`tags.map(|tag| tag).collect()` is mapping but not changing the items, is this just to get a vector? There must be a better way.

Reply:
🤦  almost certainly.